### PR TITLE
highlights 2: electric boogaloo

### LINKED ISSUE #2339: Missing highlights for hack
### Describe the highlighting problem

Hack is still missing some highlights. In particular `require` and `<?hh`. Maybe @4e554c4c could create a PR as he knows the semantics of this language.

### Example snippet that causes the problem

```hack
<?hh
require 'vendor/autoload.php';

use Facebook\HackCodegen\HackCodegenFactory;
```

### Tree-sitter parsing result

```
expression_statement [1, 0] - [1, 30]
  require_expression [1, 0] - [1, 29]
    string [1, 8] - [1, 29]
use_statement [3, 0] - [3, 44]
  use_clause [3, 4] - [3, 43]
    qualified_identifier [3, 4] - [3, 43]
      identifier [3, 4] - [3, 12]
      identifier [3, 13] - [3, 24]
      identifier [3, 25] - [3, 43]
```

### Example screenshot

image

### Expected behavior

I don't know how `require` should be highlighted and what it exactly does. Also `<?hh` is missing a highlight, maybe `@tag` 

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.2 (691469c7833068225aed48ef66b1fb4eb6f9eb38) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v17.4.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `/usr/lib/ccache/clang-14` executable found. Selected from { "/usr/lib/ccache/clang-14", "cc", "gcc", "clang", "cl", "zig" }
    Version: Ubuntu clang version 14.0.0-++20220120104511+782c0dd1a1c2-1~exp1~20220120104610.241
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - fortran        ✓ . ✓ ✓ . 
  - fusion         ✓ ✓ ✓ ✓ . 
  - ninja          ✓ . ✓ ✓ . 
  - ledger         ✓ . ✓ ✓ ✓ 
  - cmake          ✓ . ✓ . . 
  - sparql         ✓ ✓ ✓ ✓ ✓ 
  - pioasm         ✓ . . . ✓ 
  - hjson          ✓ ✓ ✓ ✓ ✓ 
  - gowork         ✓ . . . ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - dart           ✓ ✓ . ✓ ✓ 
  - http           ✓ . . . ✓ 
  - prisma         ✓ . . . . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - rasi           ✓ ✓ ✓ ✓ . 
  - cuda           ✓ ✓ ✓ ✓ ✓ 
  - hack           ✓ . . . . 
  - swift          ✓ ✓ . . . 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - java           ✓ ✓ . ✓ ✓ 
  - markdown       ✓ . ✓ . ✓ 
  - surface        ✓ . ✓ ✓ ✓ 
  - hcl            ✓ . ✓ ✓ ✓ 
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - heex           ✓ . ✓ ✓ ✓ 
  - tlaplus        ✓ . ✓ . ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 
  - ocaml_interface✓ ✓ ✓ . ✓ 
  - glimmer        ✓ . . . . 
  - scss           ✓ . . ✓ . 
  - c_sharp        ✓ ✓ ✓ . ✓ 
  - tsx            ✓ ✓ ✓ ✓ ✓ 
  - pug            ✓ . . . ✓ 
  - clojure        ✓ ✓ ✓ . ✓ 
  - norg           ✓ . ✓ . ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - supercollider  ✓ ✓ ✓ ✓ ✓ 
  - foam           ✓ ✓ ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - pascal         ✓ ✓ ✓ ✓ ✓ 
  - scala          ✓ . ✓ . ✓ 
  - yang           ✓ . ✓ . . 
  - make           ✓ . . . ✓ 
  - llvm           ✓ . . . . 
  - hocon          ✓ . . . ✓ 
  - json5          ✓ . . . ✓ 
  - nix            ✓ ✓ ✓ . ✓ 
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - vim            ✓ ✓ . . ✓ 
  - d              ✓ . ✓ ✓ ✓ 
  - haskell        ✓ . . . ✓ 
  - zig            ✓ . ✓ ✓ ✓ 
  - bibtex         ✓ . ✓ ✓ . 
  - teal           ✓ ✓ ✓ ✓ ✓ 
  - r              ✓ ✓ . . . 
  - latex          ✓ . ✓ . ✓ 
  - ql             ✓ ✓ . ✓ ✓ 
  - rst            ✓ ✓ . . ✓ 
  - fish           ✓ ✓ ✓ ✓ ✓ 
  - dot            ✓ . . . ✓ 
  - dockerfile     ✓ . . . ✓ 
  - svelte         ✓ . ✓ ✓ ✓ 
  - go             ✓ ✓ ✓ ✓ ✓ 
  - php            ✓ ✓ ✓ ✓ ✓ 
  - fennel         ✓ ✓ . . ✓ 
  - jsdoc          ✓ . . . . 
  - perl           ✓ . . . . 
  - query          ✓ ✓ ✓ ✓ ✓ 
  - godot_resource ✓ ✓ ✓ . . 
  - regex          ✓ . . . . 
  - comment        ✓ . . . . 
  - gdscript       ✓ ✓ . . ✓ 
  - verilog        ✓ ✓ ✓ . ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - ocaml          ✓ ✓ ✓ . ✓ 
  - beancount      ✓ . ✓ . . 
  - gomod          ✓ . . . ✓ 
  - erlang         . . . . . 
  - turtle         ✓ ✓ ✓ ✓ ✓ 
  - ruby           ✓ ✓ ✓ ✓ ✓ 
  - devicetree     ✓ ✓ ✓ ✓ ✓ 
  - kotlin         ✓ ✓ ✓ . ✓ 
  - elm            ✓ . . . ✓ 
  - commonlisp     ✓ ✓ ✓ . . 
  - html           ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - julia          ✓ ✓ ✓ ✓ ✓ 
  - glsl           ✓ ✓ ✓ ✓ ✓ 
  - jsonc          ✓ ✓ ✓ ✓ ✓ 
  - graphql        ✓ . . ✓ ✓ 
  - elixir         ✓ ✓ ✓ ✓ ✓ 
  - vue            ✓ . ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

```

### Output of `nvim --version`

```text
NVIM v0.7.0-dev+916-gc209f35de-dirty
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by stephan@stephan-Z87-DS3H

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info

```

### Additional context

_No response_

### LINKED ISSUE #2713: TOML: tables not highlighted separately
### Describe the highlighting problem

TOML [tables](https://toml.io/en/v1.0.0#table) are assigned the `@property` class, which they share with keys. Visual separation of the two would be desirable.

### Example snippet that causes the problem

```toml
[section]
property = "value"
number = 42
```

### Tree-sitter parsing result

```
table [0, 0] - [3, 0]
  "[" [0, 0] - [0, 1]
  bare_key [0, 1] - [0, 8]
  "]" [0, 8] - [0, 9]
  pair [1, 0] - [1, 18]
    bare_key [1, 0] - [1, 8]
    "=" [1, 9] - [1, 10]
    string [1, 11] - [1, 18]
      """ [1, 11] - [1, 12]
      """ [1, 17] - [1, 18]
  pair [2, 0] - [2, 11]
    bare_key [2, 0] - [2, 6]
    "=" [2, 7] - [2, 8]
    integer [2, 9] - [2, 11]
```

### Example screenshot

toml-table-treesitter

### Expected behavior

The vim-native syntax would classify above TOML `[section]` separately from its keys. The brackets are part of the object, which certainly gives more visual clarity. The group is linked to `hl-Title` by default, another good choice imo, considering how TOML is usually utilized (for human consumption).

toml-table-vim-native

### LINKED ISSUE #3236: Python builtin type not highlighted correctly when defining type alias.
### Describe the highlighting problem

When define type alias using builtin functions, the builtins (str, dict, int,...) is neither highlighted as "TSType" (second to last line) or "TSFuncBuiltin" (last line), but only recognised as "TSVariable".

 

### Example snippet that causes the problem

```python
from typing import Dict, TypeAlias, TypeVar

### str, int, dict is only registered as TSVariable
OldType = Dict[str, int]   # Dict highlighted correctly as TSType
T = TypeVar('T', int, float, complex)
NewType: TypeAlias = dict[str, int]


# Correct 
def func(old_var: OldType, adict: dict[str, int]):  ## TSType
    return int(3.0)          # TSFuncBuiltin
```

### Tree-sitter parsing result

Screenshot 2022-07-30 at 13 28 54
`

### Example screenshot

Screenshot 2022-07-30 at 13 22 25

### Expected behavior

`int, float, str, dict, ...` is highlighted as either builtin function or type, but not variable in type alias.
Example in VSCode
Screenshot 2022-07-30 at 13 31 37

### Output of `:checkhealth nvim-treesitter`

```markdown
## Installation
  - OK: `tree-sitter` found 0.20.6 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.16.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - make           ✓ . . . ✓ 
  - comment        ✓ . . . . 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - regex          ✓ . . . . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - vim            ✓ ✓ ✓ . ✓ 
  - yaml           ✓ ✓ ✓ ✓ ✓
```

### Output of `nvim --version`

```text
NVIM v0.8.0-dev-738-g02efdb4d5
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by phuc@Off

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/HEAD-02efdb4_1/share/nvim"

Run :checkhealth for more info
```

### Additional context

_No response_

### LINKED ISSUE #3431: [cpp] Recognize angle brackets <T> as delimiter, not operator
Similar to #2902, with similar reasoning (themes liking colorful operators).

I don't know if the grammar is able to differentiate between e.g. `if (a < b)` and `std::vector<int>` yet, but that would be nice.

cpp-angle-brackets

Rust uses `@puncutation.bracket`, which makes sense to me:

rust

### LINKED ISSUE #3460: [typescript] Recognize angle brackets <T> as delimiter, not operator
ts-angle-brackets

I reckon this would need to override what's inherited from the **ecma** highlights.

### LINKED ISSUE #3675: Track languages with missing `@type.qualifier` and/or `@storageclass`
- [ ] agda
- [ ] astro
- [ ] bash
- [ ] beancount
- [ ] bibtex
- [x] blueprint
- [ ] c
- [ ] c_sharp
- [ ] clojure
- [ ] commonlisp
- [ ] cooklang
- [ ] cpp
- [ ] cuda
- [ ] d
- [ ] dart
- [ ] devicetree
- [ ] dockerfile
- [ ] dot
- [ ] eex
- [ ] elixir
- [ ] elvish
- [ ] erlang
- [ ] fennel
- [ ] fish
- [ ] foam
- [ ] fusion
- [ ] Godot (gdscript)
- [ ] gleam
- [ ] Glimmer and Ember
- [ ] glsl
- [ ] go
- [ ] Godot Resources (gdresource)
- [ ] gomod
- [ ] gowork
- [ ] hcl
- [ ] heex
- [ ] hlsl
- [ ] java
- [ ] javascript
- [ ] jsdoc
- [ ] jsonnet
- [ ] julia
- [ ] kotlin
- [ ] lalrpop
- [ ] ledger
- [ ] llvm
- [ ] lua
- [ ] m68k
- [ ] make
- [ ] menhir
- [ ] meson
- [ ] ninja
- [ ] nix
- [ ] ocaml
- [ ] ocaml_interface
- [ ] ocamllex
- [ ] pascal
- [ ] perl
- [ ] php
- [ ] phpdoc
- [ ] pioasm
- [ ] prisma
- [ ] proto
- [ ] pug
- [ ] python
- [ ] ql
- [ ] qmljs
- [ ] r
- [ ] racket
- [ ] rasi
- [ ] rego
- [ ] rnoweb
- [ ] ruby
- [x] rust https://github.com/nvim-treesitter/nvim-treesitter/pull/3648
- [ ] scala
- [ ] scheme
- [ ] slint
- [ ] solidity
- [ ] sparql
- [ ] sql
- [ ] supercollider
- [ ] surface
- [ ] svelte
- [ ] swift
- [ ] teal
- [ ] tiger
- [ ] tlaplus
- [ ] tsx
- [ ] turtle
- [ ] twig
- [ ] typescript
- [ ] v
- [x] vala
- [ ] verilog
- [ ] vim
- [ ] vue
- [ ] wgsl
- [ ] yang
- [ ] zig

### PR DESCRIPTION
**Pull request by ObserverOfTime**
This is a continuation of #3562 (and #3567).

The plan is to fix some inconsistencies and use many of the lesser-used captures.

I will be updating this PR until it's complete ~~(hopefully by the end of the month)~~.
Then, I will tag all the language maintainers so that changes can be upstreamed if needed.

---

- Fixes #2339 (except `<?hh`)
- Fixes #2713
- Fixes #3236
- Fixes #3431
- Fixes #3460
- Resolves #3675
- Closes #1429
- Closes #2133
- Closes #2136
- Closes #3478
- Closes #3662

---

#### Progress
* [x] c
* [x] c_sharp
* [x] comment
* [x] cpp
* [x] css
* [x] cuda
* [x] d
* [x] dart
* [x] ecma
* [x] elixir
* [x] elm
* [x] elvish
* [x] fortran
* [x] gdscript
* [x] gitattributes
* [x] gleam
* [x] glsl
* [x] go (#3154)
* [x] graphql
* [x] hack
* [x] haskell
* [x] java
* [x] julia
* [x] kotlin
* [x] lalrpop
* [x] llvm
* [x] make
* [x] meson
* [x] ocaml
* [x] pascal
* [x] php
* [x] pioasm
* [x] python
* [x] ql
* [x] qmljs
* [x] rasi
* [x] ruby
* [x] rust (#3648)
* [x] scala
* [x] scss
* [x] solidity
* [x] swift
* [x] sxhkdrc
* [x] tiger
* [x] toml
* [x] twig
* [x] typescript
* [x] vala
* [x] verilog
* [x] vim (vigoux/tree-sitter-viml#130)
* [x] wgsl
* [x] yang
* [x] zig

### CODE REVIEW

**Comment by clason** [comment]
> I noticed that some languages had the upstream maintainer listed when they do not actually have any commits here so I changed those to the actual committer instead. I also fixed some incorrect usernames.

Thank you. The "maintained" information is unfortunately not always dependable -- either people just listed the upstream maintainers (without asking -- although some do have agreed to contribute here!), or because they stopped being responsive.

Long-term, we need to consider having dynamic maintenance tiers or somesuch.

**Comment by ObserverOfTime** [comment]
We can always punish maintainers that don't respond by unlisting them. /s

**Comment by clason** [comment]
Shouldn't be considered a punishment, but yes, we need some procedure for marking parsers with unresponsive listed maintainers as "unmaintained".

**Comment by ObserverOfTime** [comment]
> Shouldn't be considered a punishment

Yes, that was a joke. :)

File: queries/ocaml/highlights.scm
Comment (undu) [inline queries/ocaml/highlights.scm:106]: Reading the [ocaml manual](https://v2.ocaml.org/manual/typedecl.html), it looks like `nonrec` could be considered "type qualifier", other applied to only fields and methods, but not their _types_. Probably because how these properties are expressed. I would prefer these were kept as simple keywords

**Review by undu — APPROVED** [review]

File: queries/ocaml/highlights.scm
Comment (ObserverOfTime) [reply queries/ocaml/highlights.scm:106]: We've discussed renaming `@type.qualifier` to something more intuitive (https://github.com/nvim-treesitter/nvim-treesitter/pull/3648#issuecomment-1279958766, https://github.com/nvim-treesitter/nvim-treesitter/pull/3648#issuecomment-1291624844).
I think it's fine to use it as is for now (even if it's not applied to types) and it can be renamed in a later PR.

**Review by TravonteD — APPROVED** [review]
css & ruby look good, thanks for the update 👍🏾

**Comment by Hubro** [comment]
Upstreamed the YANG injection queries: https://github.com/Hubro/tree-sitter-yang/commit/34ad66d60b032d840a283ef09f96cfd3990a264a

The regex highlighting queries were already fixed upstream :man_shrugging: :slightly_smiling_face:

This prompted me to also add my indentations and folds queries to the tree-sitter-yang repository. I've been using them locally for months and they work great.

**Review by Hubro — APPROVED** [review]

**Comment by Decodetalkers** [comment]
https://github.com/nvim-treesitter/nvim-treesitter/pull/3662

Seems that you have finished my full request? Thanks

**Comment by tk-shirasaka** [comment]
Thanks for the update!
PHP and TOML look good.

**Comment by Luxed** [comment]
I think everything looks good on C#'s side.

**Review by Decodetalkers — APPROVED** [review]

**Comment by p00f** [comment]
Thanks, java looks good

**Comment by SalBakraa** [comment]
I see no issues with kotlin.

**Comment by alex-pinkus** [comment]
Thanks for this — I’ll upstream the Swift changes this weekend.

It makes me a bit uncomfortable that these are full-on replacements, though. Should I be worried about compatibility with other editors / highlighting themes? I’m aware of some folks who use emacs and have specified some tag names as-is. I might upstream these changes as duplicates, rather than replacements, but curious if that’s seen by others as an unnecessary burden.

File: queries/d/highlights.scm
Comment (nawordar) [inline queries/d/highlights.scm:210]: `synchronized` is a keyword marking a class or a block as accessible by one thread at a time (https://tour.dlang.org/tour/en/multithreading/synchronization-sharing). I'm not sure which capture fits the best, maybe `@storageclass.lifetime`?

File: queries/d/highlights.scm
Comment (nawordar) [inline queries/d/highlights.scm:206]: "`__gshared` stores the variable in the classic global data segment" (https://dlang.org/articles/migrate-to-shared.html#gshared). I'm not sure if `@type.qualifier` and `@storageclass.lifetime` fits better.

**Review by Prince781 — APPROVED** [review]
LGTM. Thanks.

**Comment by clason** [comment]
> Thanks for this — I’ll upstream the Swift changes this weekend.
> 
> It makes me a bit uncomfortable that these are full-on replacements, though. Should I be worried about compatibility with other editors / highlighting themes? I’m aware of some folks who use emacs and have specified some tag names as-is. I might upstream these changes as duplicates, rather than replacements, but curious if that’s seen by others as an unnecessary burden.

Yes, the queries are unfortunately editor-specific (some more than others). But that in turn means that editors don't blindly use upstream queries if they customize them, so people are free to ignore or adapt them.

I wouldn't worry too much about this -- upstream what you think makes sense and ignore the rest (or all).

File: queries/d/highlights.scm
Comment (ObserverOfTime) [reply queries/d/highlights.scm:211]: I wasn't sure about that one. I guess it should be `@type.qualifier` then?

**Comment by ambroisie** [comment]
LGTM for Tiger :ok_hand:.

File: queries/tiger/highlights.scm
Comment (ambroisie) [inline queries/tiger/highlights.scm:26]: `method` is for declaring new methods on classes, so it should be `@method` according to `CONTRIBUTING.md`.

Comment (ObserverOfTime) [reply queries/tiger/highlights.scm:26]: `@method` is for the method name.
`@keyword.function` is for the keyword that defines it.
There's no `@keyword.method` distinction.

Comment (ambroisie) [reply queries/tiger/highlights.scm:26]: Alright, makes sense to me.

File: queries/pascal/highlights.scm
Comment (Isopod) [inline queries/pascal/highlights.scm:101]: On the surface this looks reasonable, although to be honest it would personally bother me slightly if `uses` was highlighted differently from other keywords like `unit`, `interface` and `implementation`. In Pascal, a file has a very specific structure that usually goes something like this:

```pascal
unit bla;
interface
uses
  ...
type
  ...
var/const
  ...
implementation
uses
  ...
end.
```

This is almost like chapters in a book. I wouldn’t want to make one of those keywords stand out more/less than the others. You could argue that this is a styling issue and not a semantic issue – in other words: If I want them to look the same, I could just choose a theme that highlights them the same. However, I might still want to have them highlighted distinctly for other languages. So maybe there is a semantic component to it after all.

Idk, it’s not a deal-breaker or anything, might just be me 🤷

File: queries/pascal/highlights.scm
Comment (Isopod) [inline queries/pascal/highlights.scm:253]: Yes, looks good 👍

**Review by Isopod — COMMENTED** [review]
Thanks for the update. I think it would be okay to merge like this, but I did have a few comments/questions.

File: queries/pascal/highlights.scm
Comment (ObserverOfTime) [reply queries/pascal/highlights.scm:101]: You can always link `@include.pascal` to `@keyword` locally so it has the same highlights.

File: queries/pascal/highlights.scm
Comment (ObserverOfTime) [reply queries/pascal/highlights.scm:92]: I agree, the distinction is not always clear. You can see https://github.com/nvim-treesitter/nvim-treesitter/pull/3648#issuecomment-1279923861 for a comparison.

Comment (Isopod) [reply queries/pascal/highlights.scm:92]: Thanks, I already had a look at that comment earlier, but unfortunately it hasn't cleared things up for me.

- It only deals with variables/data, says nothing about functions/procedures/methods
- According to that comment, `public`, `private` etc. shouldn't be `@type.qualifier`, but you marked them as such in both C++ and Pascal (and possibly others as I haven't checked every single lanugage in this PR)

File: queries/pascal/highlights.scm
Comment (Isopod) [reply queries/pascal/highlights.scm:101]: That would be possible, but it's an extra step to take. I don't know if it is worth making a change only for users to undo it. Of course, I can't speak for everybody. Maybe others like this change.

Comment (ObserverOfTime) [reply queries/pascal/highlights.scm:101]: I think I'd rather keep it consistent with other languages which use `@include`.
If many users don't like it, we can always change it back. ¯\\_(ツ)\_/¯

**Review by zegervdv — APPROVED** [review]
Ok for the verilog changes

**Comment by szebniok** [comment]
Thanks for the update, wgsl looks good

**Comment by ckafi** [comment]
elvish maintainer here. The captures have some minor issues, but nothing too serious. I'm kinda swamped at work right now, so I can't fix it right now. I'll do it at some later date and get back to you with a PR.

File: queries/solidity/highlights.scm
Comment (YongJieYongJie) [inline queries/solidity/highlights.scm:105]: It might be more appropriate for `(yul_leave)` to be parked under `@keyword.return`, as it is used to return directly from the equivalent of inline assembly.

**Review by YongJieYongJie — COMMENTED** [review]
[Solidity] One minor comment, otherwise LGTM, and the highlighting also still looks good (at least on my colorscheme):

image

As a sidenote: I noticed the upstream maintainer of the grammar has made significant changes, I'll see if have time to port over the changes.

File: queries/gdscript/highlights.scm
Comment (theHamsta) [inline queries/gdscript/highlights.scm:23]: is it really the right thing to highlight the *whole* statement as keyword (I know it was that way before)

File: queries/toml/highlights.scm
Comment (theHamsta) [inline queries/toml/highlights.scm:4]: why type? Should then the section title be changed? what is the difference between `(pair (bare_key)) @property` and `bare_key`? I assume you want to change the highlight of the value, right?

**Comment by theHamsta** [comment]
This looks overall good. Do you think you want to handle this all in one PR (vs merging the commits individually)? Also one PR is ok. I just assume that more people need to review it to not oversee anything.

File: queries/toml/highlights.scm
Comment (ObserverOfTime) [reply queries/toml/highlights.scm:4]: `(bare_key)` is used for both table names and property names.
A top-level `(bare_key)` is a table while `(pair (bare_key))` is a property.

File: queries/julia/highlights.scm
Comment (theHamsta) [inline queries/julia/highlights.scm:121]: isn't this `@storageclass`/`@type.classifier` (I assume global/local applies to variables vs actual types)

keyword is always fine in doubt :+1:

Comment (ObserverOfTime) [reply queries/julia/highlights.scm:121]: I think they're akin to Lua's `local` and Python's `global` which are keywords.

File: queries/tiger/highlights.scm
Comment (theHamsta) [inline queries/tiger/highlights.scm:36]: probably `@operator.keyword` like in C++. `@constructor` was applied rather to identifiers constructing rather to keywords/symbols helping with construction (approximately `@function`)

**Review by theHamsta — APPROVED** [review]
This looks good :tada: ! But the diff is fairy large... Could you get another approval before merging or split up into smaller PRs?

**Comment by ObserverOfTime** [comment]
Maybe merge the changes that have been reviewed and submit individual PRs for the rest?
But let's wait a couple more days for more maintainers to respond.

**Comment by theHamsta** [comment]
Sorry didn't look at this PR a long time for the title (and because I was moving and on vacation the past three weeks). I thought this was a about a new language "boogaloo" I've never heard of :sweat_smile:

**Comment by theHamsta** [comment]
@ObserverOfTime would be fine to just have one more maintainer approval like from @clason @vigoux . I know I'm over-careful sometimes...

**Comment by ObserverOfTime** [comment]
The changes that have not yet been reviewed and I can't vouch for are dart, llvm, pioasm, and zig.

**Review by clason — APPROVED** [review]

**Comment by clason** [comment]
The only people who actually care about captures are you two, but here's an approval :)

(Things can always be tweaked in followup PRs.)

File: queries/zig/highlights.scm
Comment (theHamsta) [inline queries/zig/highlights.scm:105]: that are probably keywords and not `@attributes`. Attributes are custom annotations not added to source code elements not keywords introducing them.

Comment (theHamsta) [reply queries/zig/highlights.scm:105]: zig looks good apart from that

Comment (theHamsta) [reply queries/zig/highlights.scm:105]: llvm never turned into a real tree-sitter grammar (remove?)

Comment (theHamsta) [reply queries/zig/highlights.scm:105]: dart also looks good!

Comment (ObserverOfTime) [reply queries/zig/highlights.scm:105]: > llvm never turned into a real tree-sitter grammar (remove?)

It looks real to me. :thinking:

Comment (theHamsta) [reply queries/zig/highlights.scm]: the last time I checked it was just regex on token, without any grammar-like structure that controls their interaction. But nice when it works!

**Comment by RaafatTurki** [comment]
I see no issues with sxhkdrc, thanks for the update.

**Comment by theHamsta** [comment]
This looks good! Let's merge an clean up small details in follow-ups. Thank you for this great work :+1:

**Comment by clason** [comment]
> This looks good! Let's merge an clean up small details in follow-ups. Thank you for this great work 👍

Agreed, great work! And welcome to the team 🥳 Glad to have you on board!

**Comment by adrian5** [comment]
Fantastic effort @ObserverOfTime, bravo :clap:

**Pull request merged**
